src/apps/frontend/pages/tasks/task-modal.tsx

```diff
@@ -0,0 +1,109 @@
+import { FormikProps } from 'formik';
+import React, { useEffect } from 'react';
+
+import {
+  Button,
+  FormControl,
+  Input,
+  VerticalStackLayout,
+} from '../../components';
+import TextArea from '../../components/input/text-area';
+import Modal from '../../components/modal';
+import { ButtonKind, ButtonType } from '../../types/button';
+import { Task, TaskOperationType } from '../../types/task';
+
+interface TaskModalProps {
+  btnText: string;
+  formik: FormikProps<Task>;
+  isModalOpen: boolean;
+  setIsModalOpen: (open: boolean) => void;
+  setFieldValue: (
+    formik: FormikProps<Task>,
+    field: string,
+    value: string,
+  ) => void;
+  task?: Task;
+  taskOperationType: string;
+}
+
+const TaskModal: React.FC<TaskModalProps> = ({
+  btnText,
+  formik,
+  isModalOpen,
+  setFieldValue,
+  setIsModalOpen,
+  task,
+  taskOperationType,
+}) => {
+  useEffect(() => {
```

Comment:
I think I mentioned this, avoid using `useEffect`

Reply:
We are using the same modal for add and edit so that we can achieve reusability. useEffect will help us to check whether to open add or edit modal.
The alternative way could be different modals for add and edit if we want to avoid useEffect.

Reply:
That is not good approach either.

Are there other solutions?

Reply:
On the Modal mounting, we would like to know what type of modal it should be, i.e., "add" or "edit". However, when I tried other implementations, it was breaking our code, as it was working properly when using useEffect in the mounting of modals.